libafl/src/mutators/mutations.rs

```diff
@@ -586,14 +585,22 @@ where
             return Ok(MutationResult::Skipped);
         }
 
-        let amount = 1 + state.rand_mut().below(min(max_size - size, 16) as u64) as usize;
+        let mut amount = 1 + state.rand_mut().below(16) as usize;
         let offset = state.rand_mut().below(size as u64 + 1) as usize;
 
+        if size + amount > max_size {
+            if max_size > size {
+                amount = max_size - size;
+            } else {
+                return Ok(MutationResult::Skipped);
```

Comment:
I don't get when this will be skipped, but do we end up skipping the mutation more often?